### PR TITLE
qdq_util bug fix

### PR DESCRIPTION
**Pull request by chenfucn**
**Description**: fix a bug in quantization tool

**Motivation and Context**

when creating a temp infer file, an existing file maybe accidentally deleted

### CODE REVIEW

**Review by yufenglee — DISMISSED** [review]
:shipit:

File: onnxruntime/python/tools/quantization/quant_utils.py [hunk 1]
```diff
@@ -527,7 +528,12 @@ def model_has_infer_metadata(model):
 
 
 def load_model_with_shape_infer(model_path: Path):
-    inferred_model_path = generate_identified_filename(model_path, "-inferred")
+    temp_name_id = 0
+    while True:
+        temp_name_id += 1
+        inferred_model_path = generate_identified_filename(model_path, f"-inferred{time.time()}.{temp_name_id}")
+        if not inferred_model_path.exists():
+            break
```
Comment (justinchuby) [inline onnxruntime/python/tools/quantization/quant_utils.py:536]:
```suggestion
    for i in itertools.count():
        inferred_model_path = generate_identified_filename(model_path, f"-inferred{time.time()}.{i}")
        if not inferred_model_path.exists():
            break
```
Avoid "temp" in variable names; avoid while True loops

Helpful to also explain in a comment why this name gen process is needed?

Comment (chenfucn) [reply onnxruntime/python/tools/quantization/quant_utils.py:536]: changing it to tempfile

File: onnxruntime/python/tools/quantization/qdq_loss_debug.py [hunk 2]
```diff
@@ -344,7 +344,7 @@ def compute_signal_to_quantization_noice_ratio(
     right = numpy.concatenate(ylist).flatten()
 
     Ps = numpy.linalg.norm(left)
-    Pn = numpy.linalg.norm(left - right)
+    Pn = numpy.linalg.norm(left - right) + numpy.finfo("float").eps
```
Comment (justinchuby) [inline onnxruntime/python/tools/quantization/qdq_loss_debug.py:347]: Curious to learn why this change is needed? It would be helpful to have a comment for the readers

**Review by justinchuby — APPROVED** [review]

File: onnxruntime/python/tools/quantization/qdq_loss_debug.py (hunk 2 above)
Comment (chenfucn) [reply onnxruntime/python/tools/quantization/qdq_loss_debug.py:347]: Pn is always positive per definition of vector norm. next line Pn is the denominator, this is to avoid divide by zero problem. This is actually pretty standard in numerical programming and it was stupid of me not to include this in the first version
